src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/INamedTypeSymbolExtensions.cs

```diff
@@ -619,5 +619,11 @@ private static bool IsEqualsObject(ISymbol member)
 
         public static INamedTypeSymbol TryConstruct(this INamedTypeSymbol type, ITypeSymbol[] typeArguments)
             => typeArguments.Length > 0 ? type.Construct(typeArguments) : type;
+
+        public static bool IsValidTopLevelStatementsType(this INamedTypeSymbol type)
```

Comment:
>IsValidTopLevelStatementsType [](http://example.com/codeflow?start=27&length=29)

Based on the way the helper is used, it feels unnecessary expensive. I think there are better alternatives to achieve the goal and one of them was suggested during previous review pass.